### PR TITLE
[docs] Migrate Selection pages

### PR DESCRIPTION
**Pull request by oliviertassinari**
One more page https://deploy-preview-248--material-ui-x.netlify.app/components/data-grid/selection/

### CODE REVIEW

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 1]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
   onCellHover?: (param: CellParams) => void;
   /**
-   * Handler triggered when the click event comes from a row container element.
+   * Callback fired when the click event comes from a row container element.
    * @param param With all properties from [[RowParams]].
    */
   onRowClick?: (param: RowParams) => void;
   /**
-   * Handler triggered when the hover event comes from a row container element.
+   * Callback fired when the hover event comes from a row container element.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:194]:
```suggestion
   * Callback fired when a hover event comes from a row container element.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 2]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
    * @param param With all properties from [[RowParams]].
    */
   onRowHover?: (param: RowParams) => void;
   /**
-   * Handler triggered when one row get selected.
+   * Callback fired when one row get selected.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:199]:
```suggestion
   * Callback fired when one row is selected.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 3]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
    * @param param With all properties from [[RowSelectedParams]].
    */
   onRowSelected?: (param: RowSelectedParams) => void;
   /**
-   * Handler triggered when one or multiple rows get their selection state change.
+   * Callback fired when one or multiple rows get their selection state change.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:204]:
```suggestion
   * Callback fired when one or multiple rows get are selected.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 4]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
    * @param param With all properties from [[SelectionChangeParams]].
    */
   onSelectionChange?: (param: SelectionChangeParams) => void;
   /**
-   * Handler triggered when the click event comes from a column header element.
+   * Callback fired when the click event comes from a column header element.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:209]:
```suggestion
   * Callback fired when a click event comes from a column header element.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 5]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
    * @param param With all properties from [[ColParams]].
    */
   onColumnHeaderClick?: (param: ColParams) => void;
   /**
-   * Handler triggered when the sort model change and before a column is sorted.
+   * Callback fired when the sort model change and before a column is sorted.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:214]:
```suggestion
   * Callback fired when the sort model changes before a column is sorted.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 6]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
    * @param param With all properties from [[SortModelParams]].
    */
   onSortModelChange?: (params: SortModelParams) => void;
   /**
-   * Handler triggered when the current page has change.
+   * Callback fired when the current page has change.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:219]:
```suggestion
   * Callback fired when the current page has changed.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 7]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
    * @param param With all properties from [[PageChangeParams]].
    */
   onPageChange?: (param: PageChangeParams) => void;
   /**
-   * Handler triggered when the page size change.
+   * Callback fired when the page size change.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:224]:
```suggestion
   * Callback fired when the page size has changed.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 8]
```diff
@@ -176,57 +176,57 @@ export interface GridOptions {
...
    * @param param With all properties from [[PageChangeParams]].
    */
   onPageSizeChange?: (param: PageChangeParams) => void;
   /**
-   * Handler triggered when an exception is thrown in the grid or when the showError api method is called.
+   * Callback fired when an exception is thrown in the grid or when the showError api method is called.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:229]:
```suggestion
   * Callback fired when an exception is thrown in the grid, or when the `showError` API method is called.
```

**Review by mbrookes — CHANGES_REQUESTED** [review]

File: docs/src/pages/components/data-grid/rows/rows.md [hunk 9]
```diff
@@ -114,9 +114,9 @@ The grid exposes a set of methods that enables all of these features using the i
 
 > ⚠️ Only use this API when you have no alternative. Always start from the declarative API that the grid exposes.
 
-- `getSortModel`: Get the sort model currently applied in the grid.
-- `setSortModel`: Set the sort model of the component and trigger a new sorting of rows.
-- `onSortModelChange`: Callback fired when the columns sorting changed before the grid has sorted its rows.
+- `getSortModel`: Get the sort model currently applied to the grid.
+- `setSortModel`: Set the sort model and trigger the sorting of rows.
+- `onSortModelChange`: Callback fired when the column sorting changed before the grid has sorted its rows.
```
Comment (mbrookes) [inline docs/src/pages/components/data-grid/rows/rows.md:119]: "before the grid has sorted its rows"

Is this referring to row sort order, or column sort order within rows?

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 10]
```diff
@@ -176,27 +176,27 @@ export interface GridOptions {
...
   onRowClick?: (param: RowParams) => void;
   /**
-   * Callback fired when the hover event comes from a row container element.
+   * Callback fired when a hover event comes from a row container element.
    * @param param With all properties from [[RowParams]].
    */
   onRowHover?: (param: RowParams) => void;
   /**
-   * Callback fired when one row get selected.
+   * Callback fired when one row is selected.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:199]: One, and only one row?

File: docs/src/pages/components/data-grid/rows/rows.md (hunk 9 above)
Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/rows/rows.md:119]: What's the difference?

I think that we should remove the reference to this. There is one state, it controls how the header communicate to the users the sort order and how the rows are sorted. This state is displayed to the user in a best effort way. Sorting the rows can take a bit of time.

```suggestion
	- `onSortModelChange`: Callback fired when the column sorting changes.
```

Comment (mbrookes) [reply docs/src/pages/components/data-grid/rows/rows.md:119]: > What's the difference?

One is rows, the other is columns, so quite a significant difference!

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx (hunk 10 above)
Comment (oliviertassinari) [reply packages/grid/x-grid-modules/src/models/gridOptions.tsx:199]: I propose we remove this prop in https://github.com/mui-org/material-ui-x/issues/247

File: docs/src/pages/components/data-grid/rows/rows.md (hunk 9 above)
Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/rows/rows.md:119]: I'm confused, it's the same?

Comment (mbrookes) [reply docs/src/pages/components/data-grid/rows/rows.md:119]: The description is ambiguous, but if the prop is being removed, it doesn't matter.

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/rows/rows.md:119]: I don't think that we should remove this prop.

**Pull request merged**